### PR TITLE
Arch core concepts: docker containers, images, registries

### PR DESCRIPTION
**Pull request by sosiouxme**

### CODE REVIEW

**Comment by sosiouxme** [comment]
@adellape for your consideration

File: architecture/core_objects/images_containers.adoc [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+== Containers
+
+The basic units of OpenShift applications are called containers. Linux
+container technologies are lightweight mechanisms for isolating
+running processes so that they are limited to interacting with only
+their designated resources. Many application instances can be running
+in containers on a single host without visibility into each others'
+processes, files, network, and so on. Typically we expect that each will
+provide a single service (often called a "micro-service"), such as a
+web server or a database, though they can be used for arbitrary workloads.
```
Comment (bfallonf) [inline architecture/core_objects/images_containers.adoc]: each = ambiguous. Also, "we" might have to be avoided.

Suggest: "Typically, each container provides a single service..."

File: architecture/core_objects/images_containers.adoc [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+You can read further about container technology
+link:https://access.redhat.com/articles/1353593[here]. The Linux
+kernel has been incorporating capabilities for container
+technologies for years. More recently the Docker project has
+developed a convenient management interface for Linux containers
+on a host (link:https://www.docker.com/whatisdocker/[read more about
+Docker]). OpenShift and Kubernetes orchestrate Docker containers across
```
Comment (bfallonf) [inline architecture/core_objects/images_containers.adoc]: This link could probably be moved to the "Docker project" bit before this, and the bit in the ()s removed.

File: architecture/core_objects/images_containers.adoc [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+multi-host installations.
+
+Though you will not directly interact with Docker when using OpenShift,
+you will need to know some of the terminology and understand a little
+about how containers work in order to ensure your applications function
+well inside of containers.
```
Comment (bfallonf) [inline architecture/core_objects/images_containers.adoc]: This para could probably be incorporated into the next para, and maybe a bit more straightforward:

Suggest: "Understanding Docker terminology is important to understand its role in OpenShift and how your applications function inside of containers."

File: architecture/core_objects/images_containers.adoc [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+builders that assist with creating an image by adding your code or
+configuration to existing images.
+
+Since applications develop over time, a single image name can actually
+refer to many different versions of the "same" image. Each different
+image is referred to uniquely by its hash (a long hexadecimal number
+e.g. `fd44297e2ddb050ec4f...`)  which is usually shortened to 12
+characters (e.g. `fd44297e2ddb`). Rather than version numbers, Docker
+allows applying tags (such as `v1`, `v2.1`, `GA`, or the default `latest`)
+in addition to the image name to further specify the image desired. So
```
Comment (bfallonf) [inline architecture/core_objects/images_containers.adoc]: s/desired. So/desired, so

**Comment by bfallonf** [comment]
@sosiouxme @adellape Hope I'm not jumping the gun, but I made a few readability-wise comments. Overall, I think it's great and will do wonders when this little restructure is all over.

**Comment by sosiouxme** [comment]
Appreciate the feedback, will edit a bit!

**Comment by adellape** [comment]
@sosiouxme Close this since it's been pulled in to https://github.com/openshift/openshift-docs/pull/550?

**Pull request closed without merging**